File: Vivek/TOP_50_LEVEL1/Peak element.cpp
// Question Link : https://practice.geeksforgeeks.org/problems/peak-element/1#

// { Driver Code Starts
//Initial Template for C

#include<stdio.h>
#include<stdbool.h>


 // } Driver Code Ends
//User function Template for C

int peakElement(int arr[], int n)
{   
  // Your code here
  
  if(n==1 || (arr[0]>=arr[1]))
            return 0;
    if(n==2 && arr[1]>=arr[0])
            return 1;
   
   for(int i=1; i<n-1;i++)
   { 
       if(arr[i]>=arr[i-1] && arr[i]>=arr[i+1])
                return i ; 
   }
   
}

// { Driver Code Starts.

int main() {
	int t;
	scanf("%d", &t);
	while(t--)
	{
		int n;
		scanf("%d", &n);
		int a[n], tmp[n];
		for(int i=0;i<n;i++)
		{
			scanf("%d", &a[i]);
			tmp[i] = a[i];
		}
		bool f=0;
		
		int A = peakElement(tmp,n);
		
		if(A<0 && A>=n)
		    printf("0\n");
		else
		{
		if(n==1 && A==0)
		f=1;
		else
		if(A==0 && a[0]>=a[1])
		f=1;
		else if(A==n-1 && a[n-1]>=a[n-2])
		f=1;
		else if(a[A]>=a[A+1] && a[A]>= a[A-1])
		f=1;
		else
		f=0;
		printf("%d\n", f);
		}
		
	}

	return 0;
}  // } Driver Code Ends

File: Saurabh Kr. Vigyananand/Array GFG/Peak Element.py
# your task is to complete this function
# function should return index to the any valid peak element
class Solution:   
    def peakElement(self,arr, n):
        if(n==1):
          return 0;
        elif(arr[0]>arr[1]):
            return 0;
        elif(arr[n-1]>arr[n-2]):
            return n-1;
        else:
            for i in range(1,len(arr)-1):
                if (arr[i]>arr[i-1] and arr[i]>arr[i+1]):
                    return i

#{ 
#  Driver Code Starts
if __name__=='__main__':
    t = int(input())
    for i in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        index = Solution().peakElement(arr.copy(), n)
        flag = False
        if index<0 or index>=n:
            flag = False
        else:
            if index == 0 and n==1:
                flag = True
            elif index==0 and arr[index]>=arr[index+1]:
                flag = True
            elif index==n-1 and arr[index]>=arr[index-1]:
                flag = True
            elif arr[index-1] <= arr[index] and arr[index] >= arr[index+1]:
                flag = True
            else:
                flag = False

        if flag:
            print(1)
        else:
            print(0)
# Contributed by: Harshit Sidhwa

# } Driver Code Ends
